### ISSUE TITLE
Constructor/conversion for dimensionless quantities

### ISSUE THREAD

**Issue by sostock** [PainterQubits/Unitful.jl#358]
For pretty much all types `T <: Number`, calling `T(x)` and `convert(T, x)` are equivalent. However, for dimensionless unitful quantities, this is not the case:
```julia
julia> Float64(5u"mm/m")
5.0 mm m^-1

julia> convert(Float64, 5u"mm/m")
0.005
```
I think this is confusing and should at least be documented.

I also think that it’s very unfortunate that the `Float64` constructor does not actually return a value of type `Float64`. I would rather have a dedicated function for converting the numerical value of a quantity to a different type. Of course, this can only be changed in 2.0.

**Comment by ExpandingMan** [comment]
I strongly agree that `T(x)` should return a `T`.  This is a widely adhered to convention within the Julia ecosystem.  Changing this I believe would solve the originally posted issue.  Is there any realistic prospect of a 2.0?

**Comment by cstjean** [comment]
It might contribute to invalidations as well, by breaking `Float64`'s expected return type.
